builder/llpcTargetInfo.h

```diff
@@ -0,0 +1,164 @@
+    uint32_t gsOnChipMaxLdsSize;                // Max LDS size used by GS on-chip mode (in DWORDs)
+    uint32_t ldsSizeDwordGranularityShift;      // Amount of bits used to shift the LDS_SIZE register field
+
+    //TODO: Setup gsPrimBufferDepth from hardware config option, will be done in another change.
+    uint32_t gsPrimBufferDepth;     // Comes from the hardware GPU__GC__GSPRIM_BUFF_DEPTH configuration option
+
+    uint32_t maxUserDataCount;                  // Max allowed count of user data SGPRs
+    uint32_t tessOffChipLdsBufferSize;          // Off-chip Tess Buffer Size
+    uint32_t maxSgprsAvailable;                 // Number of max available SGPRs
+    uint32_t maxVgprsAvailable;                 // Number of max available VGPRs
+    uint32_t tessFactorBufferSizePerSe;         // Size of the tessellation-factor buffer per SE, in DWORDs.
+#if LLPC_BUILD_GFX10
+    bool     supportShaderPowerProfiling;       // Hardware supports Shader Profiling for Power
+    bool     supportSpiPrefPriority;            // Hardware supports SPI shader preference priority
+#endif
+};
+
+// Contains flags for all of the hardware workarounds which affect pipeline compilation.
+struct WorkaroundFlags
+{
+    union
+    {
+        struct
+        {
+            uint32_t  cbNoLt16BitIntClamp               :  1;
+            uint32_t  miscLoadBalancePerWatt            :  1;
+            uint32_t  miscSpiSgprsNum                   :  1;
+            uint32_t  shader8b16bLocalWriteCorruption   :  1;
+            uint32_t  shaderCoalesceStore               :  1;
+            uint32_t  shaderEstimateRegisterUsage       :  1;
+            uint32_t  shaderReadlaneSmrd                :  1;
+            uint32_t  shaderSmemBufferAddrClamp         :  1;
+            uint32_t  shaderSpiBarrierMgmt              :  1;
+            //
+            //
+            //
+            //
+            //
+
+            uint32_t  shaderSpiCsRegAllocFragmentation  :  1;
+            uint32_t  shaderVcczScalarReadBranchFailure :  1;
+            uint32_t  shaderZExport                     :  1;
+            // Pre-GFX9 hardware doesn't support min/max denorm flush, we insert extra fmul with 1.0 to flush the denorm value
+            uint32_t  shaderMinMaxFlushDenorm           :  1;
+            uint32_t  reserved                          : 19;
+        };
+        uint32_t  u32All;
+    } gfx6;
+
+    union
+    {
+        struct
+        {
+            uint32_t  fixCacheLineStraddling       :  1;
+            uint32_t  fixLsVgprInput               :  1;
+            uint32_t  shaderImageGatherInstFix     :  1;
+            uint32_t  treat1dImagesAs2d            :  1;
+            uint32_t  reserved                     : 28;
+        };
+        uint32_t  u32All;
+    } gfx9;
+
+#if LLPC_BUILD_GFX10
+    union
+    {
+        struct
+        {
+            uint32_t  disableI32ModToI16Mod        :  1;
+
+            uint32_t  waTessFactorBufferSizeLimitGeUtcl1Underflow : 1;
```

Comment:
will you add internal comments back after code merged in?

Reply:
Yes. See email I sent you.